src/libraries/Microsoft.PowerFx.Core/Public/Types/VoidType.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Types
+{
+    /// <summary>
+    /// Special Type which can't be coerced or accepted by any types.
+    /// </summary>
+    public sealed class VoidType : FormulaType
```

Comment:
Are we actually constructing these values at runtime and returning them? If so, then technically this type is "Unit", the set of one element.

Reply:
For naming, I prefer "Void" since that is very familiar to C# consumers (who will be using this API).  

We do have a placeholder value that we allocate - since we're focused to have a Value for every return type.

Reply:
Agreed, Void is a better name for our customers and users.

Reply:
Updated name to be `Void`